preflight_checks.yml

```diff
@@ -0,0 +1,314 @@
+
+- name: Store Filesystem Checks
+  set_fact:
+    preflight_results: "{{ preflight_results + [
+      {'Check': '/var is a separate partition', 'Result': filesystem_checks['var_partition']['result'], 'Reason': filesystem_checks['var_partition']['reason']},
+      {'Check': 'Root Filesystem >= 100GB', 'Result': filesystem_checks['root_fs']['result'], 'Reason': filesystem_checks['root_fs']['reason']}
+    ] }}"
+    preflight_failures: "{{ preflight_failures +
+      (['/var Partition'] if filesystem_checks['var_partition']['result'] == 'FAIL' else []) +
+      (['Root Filesystem'] if filesystem_checks['root_fs']['result'] == 'FAIL' else []) }}"
+
+- name: Extract networking facts
+  set_fact:
+    primary_nic: "{{ ansible_facts['default_ipv4']['interface'] }}"
+    primary_ip: "{{ ansible_facts['default_ipv4']['address'] }}"
+    primary_mac: "{{ ansible_facts['default_ipv4']['macaddress'] }}"
+    primary_mtu: "{{ ansible_facts[ansible_facts['default_ipv4']['interface']]['mtu'] | default('0') | int }}"
+    primary_speed: "{{ ansible_facts[ansible_facts['default_ipv4']['interface']]['speed'] | default('-1') | int }}"
+    primary_dhcp: "{{ 'dhcp' if ansible_facts['default_ipv4'].get('gateway') else 'manual' }}"
+
+- name: Define Jumbo Frames Check
+  set_fact:
+    jumbo_frames_check: "{{ 'PASS' if (primary_mtu | int) > 1500 else 'FAIL' }}"
+    jumbo_frames_reason: "{{ 'MTU is ' ~ (primary_mtu | int) ~ ', recommended > 1500' if (primary_mtu | int) <= 1500 else 'N/A' }}"
+
+- name: Store Jumbo Frames Check
+  set_fact:
+    preflight_results: "{{ preflight_results + [{'Check': 'Jumbo Frames Enabled', 'Result': jumbo_frames_check, 'Reason': jumbo_frames_reason}] }}"
+
+- name: Define NIC Configuration Check
+  set_fact:
+    nic_config_check: "{{ 'PASS' if primary_dhcp == 'manual' else 'FAIL' }}"
+    nic_config_reason: "{{ 'NIC is using DHCP, static IP is recommended' if primary_dhcp != 'manual' else 'N/A' }}"
+
+- name: Store NIC Configuration Check
+  set_fact:
+    preflight_results: "{{ preflight_results + [{'Check': 'NIC Static IP Configuration', 'Result': nic_config_check, 'Reason': nic_config_reason}] }}"
+
+- name: Define NIC Bandwidth Check
+  set_fact:
+    nic_speed_check: "{{ 'PASS' if (primary_speed | int) >= 10000 else 'FAIL' }}"
+    nic_speed_reason: "{{ 'NIC speed is ' ~ primary_speed ~ ' Mbps, recommended is 10GbE' if (primary_speed | int) < 10000 else 'N/A' }}"
+
+- name: Store NIC Bandwidth Check
+  set_fact:
+    preflight_results: "{{ preflight_results + [{'Check': 'NIC Bandwidth (10GbE Recommended)', 'Result': nic_speed_check, 'Reason': nic_speed_reason}] }}"
+
+- name: Extract NIC Details
+  set_fact:
+    nic_config_details: "{{ ansible_facts['interfaces'] }}"
```

Comment:
This task is unnecessary, just use `ansible_facts['interfaces']` directly wherever you need it
```suggestion
```